source/extensions/common/wasm/wasm.cc

```diff
@@ -1189,7 +1162,7 @@ void Context::onHttpCallResponse(uint32_t token, const Pairs& response_headers,
                                  absl::string_view response_body, const Pairs& response_trailers) {
   if (!wasm_->onHttpCallResponse_)
     return;
-  uint32_t headers_ptr, headers_size, trailers_ptr, trailers_size;
+  uint64_t headers_ptr, headers_size, trailers_ptr, trailers_size;
   exportPairs(this, response_headers, &headers_ptr, &headers_size);
   exportPairs(this, response_trailers, &trailers_ptr, &trailers_size);
   uint32_t body_ptr = wasm_->copyString(response_body);
```

Comment:
Shouldn't `body_ptr` and `body_size` be `uint64_t`?

This might be related to the problem that @bianpengyuan is seeing.

Reply:
It looks that `uint32_t` was left in a few more instances of this, namely:
- `Context::onConfigure()` casts string length to `uint32_t`,
- `Context::onRequestBody` casts body length and end-of-stream flag to `uint32_t`,
- `Context::onResponseBody` casts body length and end-of-stream flag to `uint32_t`,
- `Context::onGrpcReceive` uses `uint32_t` for `response_size` and `response_ptr`,
- `Context::onGrpcClose` uses `uint32_t` for `message_ptr`.

While `uint32_t` lengths and end-of-stream flags are probably harmless (right now), `uint32_t` pointers could be source of the real issues.

Reply:
The string lengths are unavoidable for WASM32 as the address space is < 2^32.   I don't see onGrpcRecieve using uint32_t outside of the api/wasm/cpp/*.cc file which is only included in WASM32 code and so should be uint32?  Where are you seeing that?

Reply:
In `source/extensions/common/wasm/wasm.cc` in the functions I listed in the above commit, and the `body_ptr` and `body_size` on which I commented here.

Reply:
OK, I believe I have them all.  The token is still uint32_t because it has to be but the rest are uint64_t.